### ISSUE TITLE
Paddle Conv2D 输出与 PyTorch 不一致（多组卷积参数均可复现）

### ISSUE THREAD

**Issue by tingPetty** [PaddlePaddle/Paddle#76819]
### bug描述 Describe the Bug

在对比 PyTorch 与 Paddle 的 Conv2D 时，发现多组卷积参数组合下 Paddle 的输出与 PyTorch 不一致。

下面提供一个可复现的完整代码，包含三组 Conv2D 测试用例，覆盖 stride、padding、dilation 的不同组合，均可稳定复现误差。
```
import torch
import paddle
import numpy as np

def run_conv_test(name, input_tensor, weight, bias, stride, padding, dilation, groups):
    print(f"\n======== {name} ========")

    # PyTorch Conv2D
    pt_conv = torch.nn.Conv2d(
        in_channels=weight.shape[1],
        out_channels=weight.shape[0],
        kernel_size=weight.shape[2:4],
        stride=stride,
        padding=padding,
        dilation=dilation,
        groups=groups,
        bias=True
    )
    with torch.no_grad():
        pt_conv.weight.copy_(weight)
        pt_conv.bias.copy_(bias)
        pt_out = pt_conv(input_tensor).cpu().numpy()

    # Paddle Conv2D
    paddle.set_device("cpu")
    pd_conv = paddle.nn.Conv2D(
        in_channels=weight.shape[1],
        out_channels=weight.shape[0],
        kernel_size=weight.shape[2:4],
        stride=stride,
        padding=padding,
        dilation=dilation,
        groups=groups
    )
    with paddle.no_grad():
        pd_conv.weight.set_value(weight.numpy())
        pd_conv.bias.set_value(bias.numpy())
        pd_out = pd_conv(paddle.to_tensor(input_tensor.numpy())).numpy()

    # Compare
    try:
        np.testing.assert_allclose(pt_out, pd_out, rtol=1e-6, atol=1e-5)
        print(f"[SUCCESS] PyTorch vs Paddle match.")
    except AssertionError as e:
        print(f"[FAILED] PyTorch vs Paddle differ:\n{e}")


# ==== Test Cases ====

run_conv_test(
    "test406",
    torch.randn([1, 3, 224, 224], dtype=torch.float32),
    torch.randn([64, 3, 11, 11], dtype=torch.float32),
    torch.randn([64], dtype=torch.float32),
    stride=(4, 4), padding=(2, 2), dilation=(1, 1), groups=1,
)

run_conv_test(
    "test474",
    torch.randn([1, 384, 13, 13], dtype=torch.float32),
    torch.randn([256, 384, 3, 3], dtype=torch.float32),
    torch.randn([256], dtype=torch.float32),
    stride=(1, 1), padding=(1, 1), dilation=(1, 1), groups=1,
)

run_conv_test(
    "test1269",
    torch.randn([1, 20, 10, 10], dtype=torch.float32),
    torch.randn([20, 20, 5, 5], dtype=torch.float32),
    torch.randn([20], dtype=torch.float32),
    stride=(1, 1), padding=(0, 0), dilation=(1, 1), groups=1,
)
```

期望：两者计算结果应该一致  
实际：Paddle 结果与 PyTorch 存在显著差异
实际输出结果如下：
```
信息: 用提供的模式无法找到文件。
D:\anaconda3\envs\tf_env\lib\site-packages\paddle\utils\cpp_extension\extension_utils.py:718: UserWarning: No ccache found. Please be aware that recompiling all source files may be required. You can download and install ccache from: https://github.com/ccache/ccache/blob/master/doc/INSTALL.md
  warnings.warn(warning_message)

======== test406 ========
[FAILED] PyTorch vs Paddle differ:

Not equal to tolerance rtol=1e-06, atol=1e-05

Mismatched elements: 1156 / 193600 (0.597%)
Max absolute difference: 6.484985e-05
Max relative difference: 0.0214678
 x: array([[[[ 7.339594e+00,  6.314250e+00,  1.126950e+00, ...,
          -3.920135e+00, -2.028445e+01,  7.683980e+00],
         [-2.637319e+00,  2.114627e+01, -1.385752e+01, ...,...
 y: array([[[[ 7.339595e+00,  6.314260e+00,  1.126949e+00, ...,
          -3.920135e+00, -2.028445e+01,  7.683977e+00],
         [-2.637320e+00,  2.114627e+01, -1.385752e+01, ...,...

======== test474 ========
[FAILED] PyTorch vs Paddle differ:

Not equal to tolerance rtol=1e-06, atol=1e-05

Mismatched elements: 2766 / 43264 (6.39%)
Max absolute difference: 0.00015259
Max relative difference: 0.00176154
 x: array([[[[   8.594436,  -40.519173,   11.129607, ...,  -30.900257,
           -49.044987,  -48.581875],
         [  93.89916 ,   -5.261848,  106.98966 , ...,   59.32542 ,...
 y: array([[[[   8.594432,  -40.51917 ,   11.12961 , ...,  -30.900259,
           -49.044987,  -48.581867],
         [  93.89917 ,   -5.261846,  106.98966 , ...,   59.325436,...

======== test1269 ========
[FAILED] PyTorch vs Paddle differ:

Not equal to tolerance rtol=1e-06, atol=1e-05

Mismatched elements: 13 / 720 (1.81%)
Max absolute difference: 5.340576e-05
Max relative difference: 0.00020455
 x: array([[[[-4.251427e+01,  2.359114e+01,  3.064363e+01, -1.651927e+00,
           5.262685e-01, -3.383109e+01],
         [-6.582143e-01, -8.234007e+00, -2.367112e+01, -2.293408e+01,...
 y: array([[[[-4.251429e+01,  2.359117e+01,  3.064364e+01, -1.651921e+00,
           5.262644e-01, -3.383111e+01],
         [-6.582146e-01, -8.234001e+00, -2.367110e+01, -2.293407e+01,...
```

### 其他补充信息 Additional Supplementary Information

测试环境：
- Python 3.10
- PyTorch 2.7.1
- PaddlePaddle 3.2.0 (CPU)
- OS：Windows 11

**Comment by thomasyugit** [comment]
我这里碰到了相似的问题，CUDA12.6上conv2d会有不一致，但cuda降到11.8就没有误差了，不知道是什么原因

CUDA12.6
```
paddle version: 3.2.2
torch version: 2.6.0+cu126
W1215 21:08:15.950574  7514 gpu_resources.cc:114] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 13.0, Runtime API Version: 12.6
paddle_model: Conv2D(8, 8, kernel_size=[1, 1], data_format=NCHW)
torch_model: Conv2d(8, 8, kernel_size=(1, 1), stride=(1, 1), bias=False)
dummy_input shape: (1, 8, 24, 160)
paddle_output shape: [1, 8, 24, 160]
torch_output shape: torch.Size([1, 8, 24, 160])
np.allclose: False
np.abs mean: 0.0002425048
```

CUDA11.8
```
paddle version: 3.2.2
torch version: 2.3.1+cu118
W1215 21:19:15.182729  9809 gpu_resources.cc:114] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 13.0, Runtime API Version: 11.8
paddle_model: Conv2D(8, 8, kernel_size=[1, 1], data_format=NCHW)
torch_model: Conv2d(8, 8, kernel_size=(1, 1), stride=(1, 1), bias=False)
dummy_input shape: (1, 8, 24, 160)
W1215 21:19:15.767860  9809 gpu_resources.cc:243] WARNING: device: 0. The installed Paddle is compiled with CUDNN 8.9, but CUDNN version in your machine is 8.9, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
paddle_output shape: [1, 8, 24, 160]
torch_output shape: torch.Size([1, 8, 24, 160])
np.allclose: True
np.abs mean: 0.0
```
